corehq/messaging/smsbackends/telerivet/templates/telerivet/telerivet_setup.html

```diff
@@ -0,0 +1,213 @@
+            .when('/step3', {templateUrl: 'step3.tpl'})
+            .when('/finish', {templateUrl: 'finish.tpl'})
+            .otherwise({redirectTo: '/start'});
+    });
+
+    telerivetSetupApp.controller('TelerivetSetupController', function($scope, djangoRMI) {
+        // model attributes
+        $scope.apiKey = globalApiKey;
+        $scope.projectId = globalProjectId;
+        $scope.phoneId = globalPhoneId;
+        $scope.testPhoneNumber = globalTestPhoneNumber;
+        $scope.name = '';
+        $scope.setAsDefault = '';
+
+        // error messages
+        $scope.apiKeyError = null;
+        $scope.projectIdError = null;
+        $scope.phoneIdError = null;
+        $scope.testPhoneNumberError = null;
+        $scope.nameError = null;
+        $scope.setAsDefaultError = null;
+
+        // screenshot toggles
+        $scope.showApiKeyGeneration = false;
+        $scope.showApiInfoLocation = false;
+        $scope.showAddWebookNavigation = false;
+        $scope.showWebhookDetails = false;
+
+        // control flow variables
+        $scope.showOutboundTroubleshoot = false;
+        $scope.showInboundTroubleshoot = false;
+        $scope.testSMSSent = globalTestSMSSent;
+        $scope.pollForInboundSMS = false;
+        $scope.pollingErrorOccurred = false;
+        $scope.inboundSMSReceived = false;
+        $scope.inboundWaitTimedOut = false;
+        $scope.setupComplete = false;
+
+        $scope.toggleApiInfoLocation = function() {
+            $scope.showApiInfoLocation = !$scope.showApiInfoLocation;
+        };
+
+        $scope.toggleApiKeyGeneration = function() {
+            $scope.showApiKeyGeneration = !$scope.showApiKeyGeneration;
+        };
+
+        $scope.toggleAddWebookNavigation = function() {
+            $scope.showAddWebookNavigation = !$scope.showAddWebookNavigation;
+        };
+
+        $scope.toggleWebhookDetails = function() {
+            $scope.showWebhookDetails = !$scope.showWebhookDetails;
+        };
+
+        $scope.sendTestSMS = function() {
+            djangoRMI.send_sample_sms({
+                api_key: $scope.apiKey,
+                project_id: $scope.projectId,
+                phone_id: $scope.phoneId,
+                test_phone_number: $scope.testPhoneNumber,
+                request_token: '{{ request_token }}',
+            })
+            .success(function(data) {
+                $('#id_send_sms_button')
+                .text('{% trans "Send" %}');
+                $scope.apiKeyError = data.api_key_error;
+                $scope.projectIdError = data.project_id_error;
+                $scope.phoneIdError = data.phone_id_error;
+                $scope.testPhoneNumberError = data.unexpected_error || data.test_phone_number_error;
+                $scope.testSMSSent = data.success;
+            })
+            .error(function() {
+                $('#id_send_sms_button')
+                .text('{% trans "Server error. Try again..." %}');
```

Comment:
you can use javascripts translations here instead of hardcoding the template ones: `gettext("Server error. Try again...")`